File: solace/cmd/cooldown/CoupDeGrace.java
package solace.cmd.cooldown;

import java.util.*;
import solace.game.*;
import solace.net.*;
import java.io.*;
import solace.util.*;
import solace.cmd.InvalidTargetException;

/**
 * "Coup de Grace" is a cooldown action that does 1000 potency damage to a
 * target whos health is at 30% or less.
 * @author Ryan Sandor Richards
 */
public class CoupDeGrace extends CooldownCommand {
  public CoupDeGrace(Player p) {
    super("coup", p);
    setDisplayName("coup de grace");
    setCooldownDuration(120);
    setInitiatesCombat(true);
    setBasePotency(1000);
    addResourceCost(new SpCost(10));
  }

  protected void checkValidTarget(Player target) throws InvalidTargetException {
    super.checkValidTarget(target);
    boolean isValid = (double)target.getHp() / (double)target.getMaxHp() < 0.3;
    if (!isValid) {
      throw new InvalidTargetException(String.format(
        "%s can only be used on targets with less than 30% health.",
        getDisplayName()));
    }
  }

  public boolean execute(int level, Player target) {
    return executeAttack(target);
  }
}

File: script/cooldowns/coup-de-grace.js
'use strict';

/**
 * "Coup de Grace" is a cooldown action that does 1000 potency damage to a
 * target whos health is at 30% or less.
 * @author Ryan Sandor Richards
 */
Commands.addCooldown('coup', {
  displayName: 'coup de grace',
  cooldownDuration: 120,
  initiatesCombat: true,
  basePotency: 1000,
  spCost: 10,
  run: function (player, target, level, cooldown) {
    return cooldown.executeAttack(target);
  },
  checkValidTarget: function (target) {
    if (target.getHp() / target.getMaxHp() >= 0.3) {
      throw new Error(
        'Coup de grace can only be used on targets with less than 30% health.'
      );
    }
  }
});
